File: ballast-client/dotnet/src/dependency-injection/Options.cs
using System;

namespace Ballast.Client.DependencyInjection 
{
    public interface IBallastClientOptions 
    { 
        bool UseSignalR { get; set;}
        string ServerUrl { get; set; }
        Guid? ClientId { get; set; }
    }

    public class BallastClientOptions : IBallastClientOptions 
    { 

        public bool UseSignalR { get; set; }
        public string ServerUrl { get; set; }
        public Guid? ClientId { get; set; }

        public BallastClientOptions() 
        { 
            UseSignalR = true;
            ServerUrl = null;
            ClientId = null;
        }

    }
}

File: ballast-client/npm/src/dependency-injection/options.ts
import { Guid } from "ballast-core";

export interface IBallastClientOptions {
    useSignalR: boolean;
    serverUrl: string | null;
    clientId: string | null;
}

export class BallastClientOptions implements IBallastClientOptions { 

    public useSignalR: boolean;
    public serverUrl: string | null;
    public clientId: string | null;

    public constructor() {
        this.useSignalR = true;
        this.serverUrl = null;
        this.clientId = Guid.newGuid();
    }
    
}